tests/test_expression.py

```diff
@@ -61,6 +78,316 @@ def test_safe_resolve_dict(self):
+                    "is_active": True,
+                    "list": [{"a": 1}, {"a": 2}],
+                    "dict": {"a": 3},
+                }
+            ),
+        )
+
+        # Check that variables passed to the component are of correct type
+        self.assertEqual(captured["bool_var"], True)
+        self.assertEqual(captured["dict_var"], {"a": 3})
+        self.assertEqual(captured["list_var"], [{"a": 1}, {"a": 2}])
+
+        self.assertHTMLEqual(
+            rendered,
+            """
+            <div>lorem ipsum dolor</div>
+            <div>True</div>
+            <div>[{'a': 1}, {'a': 2}]</div>
+            <div>{'a': 3}</div>
+            """,
+        )
+
+    @parametrize_context_behavior(["django", "isolated"])
+    def test_comment_in_template(self):
+        registry.library.tag(noop)
+        captured = {}
+
+        @register("test")
+        class SimpleComponent(Component):
+            def get_context_data(
+                self,
+                pos_var1: Any,
+                pos_var2: Any,
+                *args: Any,
+                bool_var: bool,
+                list_var: Dict,
+            ):
+                captured["pos_var1"] = pos_var1
+                captured["pos_var2"] = pos_var2
+                captured["bool_var"] = bool_var
+                captured["list_var"] = list_var
+
+                return {
+                    "pos_var1": pos_var1,
+                    "pos_var2": pos_var1,
+                    "bool_var": bool_var,
+                    "list_var": list_var,
+                }
+
+            template: types.django_html = """
+                <div>{{ pos_var1 }}</div>
+                <div>{{ pos_var2 }}</div>
+                <div>{{ bool_var }}</div>
+                <div>{{ list_var|safe }}</div>
+            """
+
+        template_str: types.django_html = (
+            """
+            {% load component_tags %}
+            {% component 'test'
+                "{# lorem var_a w #}"
+                " {# lorem var_a w #} abc"
+                bool_var="{# noop is_active #}"
+                list_var=" {# noop list #} "
+            / %}
+        """.replace(
+                "\n", " "
+            )
+        )
+
+        template = Template(template_str)
+        rendered = template.render(
+            Context(
+                {
+                    "var_a": 3,
+                    "is_active": True,
+                    "list": [{"a": 1}, {"a": 2}],
+                }
+            ),
+        )
+
+        # Check that variables passed to the component are of correct type
+        self.assertEqual(captured["pos_var1"], "")
+        self.assertEqual(captured["pos_var2"], "  abc")
+        self.assertEqual(captured["bool_var"], "")
+        self.assertEqual(captured["list_var"], "  ")
+
+        self.assertHTMLEqual(
+            rendered,
+            """
+            <div></div>
+            <div> </div>
+            <div></div>
+            <div> </div>
+            """,
```

Comment:
Maybe not use assertHTMLEqual here since this is testing stringification and handling of spaces?